sleuth/sleuth/enrich_tls.py

```diff
@@ -44,120 +44,58 @@
 VALID = 2
 
 
-security_categories = {
-    "unknown": 0,
-    "avoid": 1,
-    "legacy": 2,
-    "acceptable": 3,
-    "recommended": 4,
-}
+class Policy:
+    'Class to import seclevel policy'
+
+    def __init__(self, filename, failure_threshold):
+        cur_dir = os.path.dirname(__file__)
+        policy_path = os.path.join(cur_dir, filename)
+        with open(policy_path) as f:
+            policy_data = json.load(f)
+            self.classifications = policy_data["classification"]
+            self.rules = policy_data["rules"]
+            self.failure_threshold = int(failure_threshold) if failure_threshold else policy_data["default_failure_threshold"]
+
+    def seclevel(self, x):
+        seclevel_names = {v: k for k, v in self.classifications.iteritems()}
+        seclevel_names[UNKNOWN] = "unknown"
+        if x in seclevel_names:
+            return seclevel_names[x]
+        return 0
+
+
+# It should be noted that this is a soft check on whether or not the selected
+# cipher suite would be acceptable to the specified compliance policy
+#
+def check_compliance(compliance_policy, scs):
+    if not scs:
+        return 'unknown'
 
+    if not compliance_policy:
+        return 'unknown'
 
-seclevel_names = {v: k for k, v in security_categories.iteritems()}
+    cur_dir = os.path.dirname(__file__)
+    check_data_file = "data_tls_with_desc.json"
+    check_data_path = os.path.join(cur_dir, check_data_file)
 
+    compliance_data_file = "compliance.json"
+    compliance_data_path = os.path.join(cur_dir, compliance_data_file)
 
-def seclevel(x):
-    if x in seclevel_names:
-        return seclevel_names[x]
-    if x in security_categories:
-        return security_categories[x]
-    return 0
+    with open(check_data_path) as check_f:
```

Comment:
Looks like the program is failing here because data_tls_with_desc.json does not exist. Did you mean to reflect that change here?